### PR TITLE
Add Host Endpoints

### PR DESCRIPTION
**Pull request by ashcrow**
Implements:
- http://commissaire.readthedocs.io/en/latest/endpoints.html#host
- http://commissaire.readthedocs.io/en/latest/endpoints.html#hosts
- http://commissaire.readthedocs.io/en/latest/endpoints.html#hostcreds
- http://commissaire.readthedocs.io/en/latest/endpoints.html#hoststatus

### CODE REVIEW

**Comment by ashcrow** [comment]
[(8) [multi-host-mgr-2] Port host endpoints](https://trello.com/c/i2qtx8oG/486-8-multi-host-mgr-2-port-host-endpoints)

File: src/commissaire_http/handlers/hosts.py [hunk 1]
```diff
@@ -152,12 +152,23 @@ def delete_host(message, bus):
     :returns: A jsonrpc structure.
     :rtype: dict
     """
-    # TODO: kick off service job to remove the host
     try:
```
Comment (mbarnes) [inline src/commissaire_http/handlers/hosts.py]: nit: Not required, but `address = message['params']['address']` here would save some typing below.

File: src/commissaire_http/handlers/hosts.py [hunk 2]
```diff
@@ -243,3 +239,27 @@ def get_host_status(message, bus):
...
+    """
+    Shorthand to check and see if a cluster exists. If it does, return the
+    message response data.
+
+    :param bus: Bus instance.
+    :type bus: commissaire_http.bus.Bus
+    :param cluster_name: The name of the Cluster to look up.
+    :type cluster_name: str
+    :returns: The message structure describing the found Cluster or None
+    :rtype: mixed
```
Comment (mbarnes) [inline src/commissaire_http/handlers/hosts.py]: Would it be any more useful to return a `Cluster` model or `None`?

File: test/__init__.py [hunk 3]
```diff
@@ -50,6 +52,21 @@ def get_fixture_file_path(filename):
         'Can not find path for config: {0}'.format(filename))
 
 
+def expected_error(message_id, code):
+    """
+    Creats an expected error structure with the error information as mock.ANY.
```
Comment (mbarnes) [inline test/__init__.py]: typo: Creats -> Creates

**Comment by ashcrow** [comment]
Fixing up. Thanks @mbarnes!!

File: test/__init__.py (hunk 3 above)
Comment (ashcrow) [inline test/__init__.py]: Fixed

File: src/commissaire_http/handlers/hosts.py (hunk 2 above)
Comment (ashcrow) [inline src/commissaire_http/handlers/hosts.py]: Done

File: src/commissaire_http/handlers/hosts.py [hunk 4]
```diff
@@ -110,3 +112,42 @@ def get_hostcreds(message, bus):
         LOGGER.debug('Client requested a non-existant host: "{}"'.format(
             message['params']['address']))
         return return_error(message, error, JSONRPC_ERRORS['NOT_FOUND'])
+
+
+def get_host_status(message, bus):
+    """
+    Gets the status of an exisiting host.
```
Comment (ashcrow) [inline src/commissaire_http/handlers/hosts.py]: Done

File: src/commissaire_http/handlers/hosts.py (hunk 1 above)
Comment (ashcrow) [inline src/commissaire_http/handlers/hosts.py]: Done :smile:

**Pull request merged**